libraries/standard/http/include/http_client.h

```diff
@@ -0,0 +1,357 @@
+     * @param[in] pContext User context.
+     * @param[in] fieldLoc Location of the header field name in the response buffer.
+     * @param[in] fieldLen Length in bytes of the field name.
+     * @param[in] valueLoc Location of the header value in the response buffer.
+     * @param[in] valueLen Length in bytes of the value.
+     * @param[in] statusCode The HTTP response status-code.
+     */
+    void (*onHeaderCallback)( void* pContext, const char* fieldLoc, size_t fieldLen, const char* valueLoc, size_t valueLen, uint16_t statusCode );
+
+    /* Private context for the application. */
+    void *pContext;
+} HTTPClient_HeaderParsingCallback_t;
+
+/**
+ * @brief Represents an HTTP response.
+ */
+typedef struct HTTPResponse
+{
+    /**
+     * @brief Buffer for both the raw HTTP header and body. 
+     * 
+     * This buffer is supplied by the application.
+     * 
+     * This buffer is owned by the library during  #HTTPClient_Send. This buffer
+     * should not be modifed until after this functions return.
+     * 
+     * For optimization this buffer may be re-used with the request. The user 
+     * can re-use this buffer for the storing the request headers in 
+     * #HTTPRequestHeaders_t.pBuffer.
+     */
+    uint8_t* pBuffer;
+    size_t bufferLen; /**< The length of the response buffer in bytes. */
+
+    /**
+     * @brief Optional callback for intercepting the header during the first 
+     * parse through of the response as is it receive from the network.
+     */
+    HTTPClient_HeaderParsingCallback_t* pHeaderParsingCallback;
+    
+    /**
+     * @brief The starting location of the response headers in pBuffer.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint8_t* pHeaders;
+
+    /**
+     * @brief Byte length of the response headers in pBuffer.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t headersLen; 
+
+    /**
+     * @brief The starting location of the response body in pBuffer.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint8_t* pBody;
+
+    /**
+     * @brief Byte length of the body in pBuffer.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t bodyLen;
+    
+    /* Useful HTTP header values found. */
+
+    /**
+     * @brief The HTTP response Status-Code.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint16_t statusCode;
+    
+    /**
+     * @brief The value in the "Content-Length" header is returned here.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t contentLength;
+
+    /**
+     * @brief Count of the headers sent by the server.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    size_t headerCount;
+
+    /**
+     * @brief Flags of useful headers found in the response.
+     * 
+     * This is updated by #HTTPClient_Send.
+     */
+    uint32_t flags;
+} HTTPResponse_t;
+
+/**
+ * TODO: Add through documentation.
```

Comment:
Spelling "thorough".  Also, why not go ahead and add some documentation here rather than a `TODO`?  It doesn't have to be complete.